pyepp/cli.py

```diff
@@ -0,0 +1,193 @@
+"""
+PyEpp command line interface module.
+"""
+# pylint: skip-file
+import functools
+import pprint
+
+import click
+
+from pyepp.helper import xml_pretty
+from pyepp.epp import EppResultData
+from pyepp.host import Host
+from pyepp.contact import Contact
+from pyepp.domain import Domain
+from pyepp import EppCommunicator
+
+
+def login_logout(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        self.connect()
+        self.login()
+        result = func(self, *args, **kwargs)
+        self.logout()
+
+        return result
+
+    return wrapper
+
+
+class PyEppCli:
+    def __init__(self, host, port, client_cert, client_key, user, password, output_format, no_pretty):
+        self.epp = EppCommunicator(host, port, client_cert, client_key)
+
+        self.user = user
+        self.password = password
+
+        self.output_format = output_format
+        self.no_pretty = no_pretty
+
+        self.registry_object = None
+
+    def connect(self):
+        self.epp.connect()
+
+    def login(self):
+        self.epp.login(self.user, self.password)
+
+    def logout(self):
+        self.epp.logout()
+
+    def format_output(self, result: EppResultData):
+        output = result
+        if self.output_format == 'XML':
+            output = result.raw_response
+            if not self.no_pretty:
+                output = xml_pretty(result.raw_response)
+        else:
+            if not self.no_pretty:
+                output = pprint.pformat(output)
+
+        return output
+
+    @login_logout
+    def check(self, *args, **kwargs):
```

Comment:
Are these args coming straight from the command line? It would be nice to check them to provide the user with useful error messages if they are missing args

Reply:
That's a good point. But have yet to reach the point to implement them. That will be when I'm implementing those functionalities.